gradio/utils.py

```diff
@@ -39,23 +46,31 @@ def version_check():
         warnings.warn("unable to connect with package URL to collect version info.")
 
 
-def initiated_analytics(data):
+def get_local_ip_address() -> str:
```

Comment:
Migrating to FastAPI, I would recommend using the async advantages of the fastapi, that way requests would be async and do not block. [Example](https://github.com/FarukOzderim/Memory-Vault/blob/151b3fc548540445221d2ff57b0338c6fce313a6/src/events.py#L119)

Reply:
Great catch!

Reply:
Looking at this again, this only gets called when `Interface()` is instantiated, which happens outside of fastapi. So no need to make this particular one async.